### PR TITLE
New-AzGalleryImageVersion new parameter SourceImageVMId

### PR DESCRIPTION
**Pull request by Sandido**
## Description

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1412
This adds a new parameter SourceImageVMId. This takes in a resourceId that is specifically for a VM. This cannot be used when the SourceImageId is also used. 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs [hunk 1]
```diff
@@ -260,6 +260,43 @@ public override void ExecuteCmdlet()
                         }
                     }
 
+                    if (this.IsParameterBound(c => c.SourceImageVMId))
+                    {
+                        if (!this.SourceImageVMId.ToLower().Contains("/virtualmachines/"))
+                        {
+                            throw new Exception("The `SourceImageVMId` parameter must be a valid virtual machine reference id such as /subscriptions/<sub>/resourceGroups/<rg>/providers/Microsoft.Compute/virtualMachines/<vm>");
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs:267]: could this type of error message come from the service level?

File: src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs [hunk 2]
```diff
@@ -260,6 +260,43 @@ public override void ExecuteCmdlet()
...
+                        }
+
+                        if (this.IsParameterBound(c => c.SourceImageId))
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs:270]: same comment as above

**Comment by vidai-msft** [comment]
@Sandido Please address the review comments.

File: src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs (hunk 2 above)
Comment (Sandido) [reply src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs:270]: Not before this PS release. This error will be added in the backend when they can.

File: src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs (hunk 1 above)
Comment (Sandido) [reply src/Compute/Compute/Generated/GalleryImageVersion/GalleryImageVersionCreateOrUpdateMethod.cs:267]: Not before this PS release. This error will be added in the backend when they can.

**Comment by Sandido** [comment]
> @Sandido Please address the review comments.

@vidai-msft , done. The errors will happen on the backend when we can get that team to work on it. But this is an important feature to get in. I will be following up to ensure we can remove these.

**Comment by Sandido** [comment]
Apparently the vmID format error is already in the backend, so removing it.

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**